contracts/src/StakeTable.sol

```diff
@@ -191,18 +217,66 @@ contract StakeTable is AbstractStakeTable {
         override
         returns (uint64, uint64)
     {
-        bytes32 hash = _hashBlsKey(blsVK);
-        nodes[hash].balance += amount;
-        return (0, 0);
+        bytes32 key = _hashBlsKey(blsVK);
+        Node memory node = nodes[key];
+
+        // The deposit must come from the node's registered account.
+        if (node.account != msg.sender) {
+            revert CallerIsNotTheNodeOwner();
+        }
+
+        // A node cannot deposit more tokens while it waiting to register,
+        if (currentEpoch() <= node.registerEpoch) {
+            revert PrematureDeposit();
+        }
+
+        // A node cannot deposit more tokens if an exit request is in progress.
+        if (node.exitEpoch != 0) {
+            revert ExitRequestInProgress();
+        }
+
+        node.balance += amount;
+        nodes[key] = node;
+        SafeTransferLib.safeTransferFrom(ERC20(tokenAddress), msg.sender, address(this), amount);
+
+        emit Deposit(_hashBlsKey(blsVK), uint256(amount));
+
+        uint64 effectiveEpoch = currentEpoch() + 1;
+
+        return (node.balance, effectiveEpoch);
     }
 
     /// @notice Request to exit from the stake table, not immediately withdrawable!
     ///
     /// @param blsVK The BLS verification key to exit
     /// @return success status
     function requestExit(BN254.G2Point memory blsVK) external override returns (bool) {
-        bytes32 hash = _hashBlsKey(blsVK);
-        nodes[hash].exitEpoch = 0;
+        bytes32 key = _hashBlsKey(blsVK);
+        Node memory node = nodes[key];
+
+        // The exit request must come from the node's withdrawal account.
+        if (node.account != msg.sender) {
+            revert CallerIsNotTheNodeOwner();
+        }
```

Comment:
again, do we need this?

Reply:
In this case I think we should not allow anyone to request an exit for a specific node, right?

Reply:
as conceded in https://github.com/EspressoSystems/espresso-sequencer/pull/791#discussion_r1416734021, I agree.